### PR TITLE
sonic-cfggen supports hwsku parameter

### PR DESCRIPTION
**Pull request by qiluo-msft**
and Install l2switch.j2 into templates directory

### CODE REVIEW

File: src/sonic-config-engine/setup.py [hunk 1]
```diff
@@ -19,4 +20,7 @@ def get_test_suite():
       scripts=['sonic-cfggen'],
       install_requires=['lxml', 'jinja2', 'netaddr', 'ipaddr', 'pyyaml', 'pyangbind'],
       test_suite='setup.get_test_suite',
+      data_files=[
+        ('/usr/share/sonic/templates', glob.glob('data/*')),
+      ],
      )
```
Comment (taoyl-ms) [inline src/sonic-config-engine/setup.py]: Why do we install this file with sonic-config-engine? Shouldn't it go with base image?

File: src/sonic-config-engine/sonic-cfggen [hunk 2]
```diff
@@ -15,6 +15,7 @@ Examples:
 See usage string for detail description for arguments.
 """
 
+from __future__ import print_function
```
Comment (taoyl-ms) [inline src/sonic-config-engine/sonic-cfggen]: What's it for? Print to stderr?

Comment (qiluo-msft) [reply src/sonic-config-engine/sonic-cfggen]: To write 'print' in the most compatible style, for example python3.

---
In reply to: [183192730](https://github.com/Azure/sonic-buildimage/pull/1631#discussion_r183192730) [](ancestors = 183192730)

File: src/sonic-config-engine/sonic-cfggen [hunk 3]
```diff
@@ -162,7 +165,18 @@ def main():
...
     data = {}
+    hwsku = args.hwsku
 
+    if hwsku is not None:
+        hardware_data = {'DEVICE_METADATA': {'localhost': {
+            'hwsku': hwsku
+            }}}
+        deep_update(data, hardware_data)
+        (ports, _) = get_port_config(hwsku, platform, args.port_config)
+        if not bool(ports):
```
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply src/sonic-config-engine/sonic-cfggen]: Fixed.

---
In reply to: [183192799](https://github.com/Azure/sonic-buildimage/pull/1631#discussion_r183192799) [](ancestors = 183192799)

**Comment by qiluo-msft** [comment]
Retest this please

**Comment by lguohan** [comment]
@qiluo-msft , can you check the test?

**Comment by qiluo-msft** [comment]
Retest this please

**Comment by qiluo-msft** [comment]
Retest this please

File: src/sonic-config-engine/tests/sample_output/l2switch.json [hunk 4]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (lguohan) [inline src/sonic-config-engine/tests/sample_output/l2switch.json]: we need management port configuration.

Comment (qiluo-msft) [reply src/sonic-config-engine/tests/sample_output/l2switch.json]: This PR focuses on hwsku related feature.

---
In reply to: [183607010](https://github.com/Azure/sonic-buildimage/pull/1631#discussion_r183607010) [](ancestors = 183607010)

**Pull request merged**